plugin/pulpcore/plugin/download/http.py

```diff
@@ -1,8 +1,34 @@
+import logging
+
 import aiohttp
+import backoff
 
 from .base import attach_url_to_exception, BaseDownloader, DownloadResult
 
 
+log = logging.getLogger(__name__)
+
+
+logging.getLogger('backoff').addHandler(logging.StreamHandler())
+
+
+def inspect_to_retry(exc):
+    """
+    Inspect a raised exception to determine if a retry is needed
+
+    Retries when the status code is one of the following:
+
+        429 - Too Many Requests
+
+    Args:
+        exc (Exception): The exception to inspect
```

Comment:
Presumably, if there is a special "status" field on the exception, there is some more specific Exception base class being expected here?